sklearn/utils/estimator_checks.py

```diff
@@ -2486,6 +2488,17 @@ def check_outliers_fit_predict(name, estimator_orig):
             assert_raises(ValueError, estimator.fit_predict, X)
 
 
+def check_fit_non_negative(name, estimator_orig):
+    # Check that proper warning is raised for non-negative X
+    # when tag requires_positive_X is present
+    X = np.array([[-1., 1], [-1., 1]])
+    y = np.array([1, 2])
+    estimator = clone(estimator_orig)
+    assert_raises_regex(ValueError, "Negative values in data passed to",
```

Comment:
We've decided to deprecate assert_raise_regex, see #14216

Reply:
I would need to import pytest in a non-test file. Do we have such a context manager in the testing module?

Reply:
I don't know, let's leave it as it is.